libr/include/r_types.h

```diff
@@ -654,4 +654,16 @@ static inline void r_run_call10(void *fcn, void *arg1, void *arg2, void *arg3, v
 # endif
 #endif
 
+// reference counter
+typedef void (*RFreeCallback)(void *a);
+typedef RRef volatile size_t;
+
+#define r_ref(x) x->refcount++;
+#define r_unref(x,f) { assert (x->refcount > 0); if (!--x->refcount) { f(x); } }
+
+#define R_REF_TYPE() RRef refcount;
+#define R_REF_FUNCTIONS(s, n, f) \
+R_API void r_##n##_ref(s *x) { r_ref (x); } \
+R_API void r_##n##_unref(s *x) { r_unref (x, f); }
```

Comment:
I think this should be `static inline`

Reply:
isnt the unit test enough as an example?